MaterialDesignThemes.Wpf/PaletteHelper.cs

```diff
@@ -123,24 +151,41 @@ public void ReplaceAccentColor(string name)
         /// <param name="newValue">The new entry value</param>
         /// <param name="parentDictionary">The root dictionary to start searching at. Null means using Application.Current.Resources</param>
         /// <returns>Weather the value was replaced (true) or not (false)</returns>
-        private static bool ReplaceEntry(object entryName, object newValue, ResourceDictionary parentDictionary = null)
+        private static bool ReplaceEntry(object entryName, object newValue, ResourceDictionary parentDictionary = null, bool animate = true)
         {
+            const int DURATION_MS = 500; //Change the value if needed
             if (parentDictionary == null)
                 parentDictionary = Application.Current.Resources;
-
+            
             if (parentDictionary.Contains(entryName))
             {
-                parentDictionary[entryName] = newValue;
+                if (animate & parentDictionary[entryName] != null & GetPowerState().ACLineStatus == ACLineStatus.Online) //Fade animation is enabled and value is not null.
+                {
+                    try
+                    {
+                        ColorAnimation animation = new ColorAnimation()
+                        {
+                            From = ((SolidColorBrush)parentDictionary[entryName]).Color,//The old color
+                            To = ((SolidColorBrush)newValue).Color, //The new color
+                            Duration = new Duration(new TimeSpan(0,0,0,0,DURATION_MS)) //Set the duration
+                        };
+                        (parentDictionary[entryName] as SolidColorBrush).BeginAnimation(SolidColorBrush.ColorProperty, animation); //Begin the animation
+                    }
+                    catch
+                    {
+                        System.Diagnostics.Debug.WriteLine(@"The argument :" + nameof(newValue) + " is not a brush.");
+                        parentDictionary[entryName] = newValue; //Set the value normally if type is incorrect
+                    }
```

Comment:
What's wrong with making a check instead of an unsafe cast and catching the exception?
```
var solidColorBrush = parentDictionary[entryName] as SolidColorBrush;
if (solidColorBrush == null) return;

var animation = new ColorAnmiation { . . . };
solidColorBrush.BeginAnimation( . . . );
```